app/lib/backend_client.rb

```diff
@@ -1,9 +1,30 @@
+# frozen_string_literal: true
+
 module BackendClient
   class BackendError < RuntimeError; end
   class TooManyItems < BackendError; end
   class InvalidItem  < BackendError; end
 
-  def self.config
-    System::Application.config.backend_client
+  class << self
+    def threescale_client_config
+      config.merge(
+        port: uri.port,
+        host: uri.host
+      )
+    end
+
+    def config
+      System::Application.config.backend_client
+    end
+
+    protected
+
+    def uri
+      URI(config[:url].presence || "#{scheme}://#{config[:host]}")
+    end
+
+    def scheme
+      config[:secure] ? 'https' : 'http'
+    end
```

Comment:
Not memoizing because they are class methoids, and it is not thread-safe ... Though the content would not likely be changed BUT the object reference would ...